File: Solutions/648. Replace Words/648.java
class Solution {
    public String replaceWords(List<String> dictionary, String sentence) {
        String[] split = sentence.split(" ");
        Collections.sort(dictionary, Comparator.comparingInt(String::length));
        for(int i=0; i< split.length; i++){
            for(String root: dictionary) {
                if(split[i].startsWith(root)){
                    split[i] = root;
                }
            }
        }
        return String.join(" ", split);
    }
}

File: Solutions/648. Replace Words/648.py
class Solution:
    def replaceWords(self, dictionary: List[str], sentence: str) -> str:
        sorted(dictionary, key=len)
        ans = sentence.split()
        for i in range(len(ans)):
            for root in dictionary:
                if ans[i].startswith(root):
                    ans [i] = root
        return ' '.join(ans)   